specifications/drtm-handler-and-table.rst

```diff
@@ -0,0 +1,310 @@
+DLE Handler Specification
+=========================
+
+This specification details the state the platform should be in when invoking a
+DLE Handler.
+
+Overview
+--------
+
+The DLE (Dynamic Launch Event) Handler consumes the TrenchBoot DRTM Table and
+invokes the platform's dynamic launch event. This specification seeks to
+standardize the invocation interface for the DLE Handler for each platform
+supported by TrenchBoot. This will provide a well-defined interface for DLE
+Handler implementors to follow and bootloader to comply to ensure
+interoperability between implementations.
+
+Acronyms
+--------
+
+DCE - Dynamic Configuration Environment (Intel ACM/AMD SLB)
+DLE - Dynamic Launch Event (Intel GETSEC[SENTER]/AMD SKINIT)
+DLE Handler - Dynamic Launch Event Handler
+DLME - Dynamic Launch Measured Environment (Operating System/Hypervisor)
+DRTM - Dynamic Root of Trust Measurement
+SL Entry - Secure Launch Entry (Secure Launch Stub for Linux)
+
+Platform Requirements
+---------------------
+
+1 - x86 Platforms
+1.1 - The DLE Handler will be invoked with the CPU in 64bit long mode 
+1.2 - The DRTM Table will be passed to the DLE Handler in the RDI CPU register
+
```

Comment:
If I understand correctly a client such as the EFI stub will invoke the DLE Handler.

- Where is the DLE Handler implemented?  Is this in firmware/GRUB, or in the kernel?  From the lkml discussion I had the impression that EFI calls back into firmware.
- If in firmware how does the client/EFI-stub know what the address of the handler is?
- Is there some kind of EFI or other calling convention assumed here about how the call is made?  Like which instruction is used, parameter registers, etc.
- Presumably in normal operation the DLE Handler should never return.  That should be stated.
- What if the DLE handler finds an error in the DRTM table passed?  Will it return an error?  Are there error codes we should define?

Reply:
New version posted.